// app/public/utils/test.browser.js
(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({"/Users/josep/Documents/work/puntoneutro/probes/utils/index.js":[function(require,module,exports){


module.exports = function(){

	return {
		extend: function( defaults, options ) {
	        var extended = {};
	        var prop;
	        for (prop in defaults) {
	            if (Object.prototype.hasOwnProperty.call(defaults, prop)) {
	                extended[prop] = defaults[prop];
	            }
	        }
	        for (prop in options) {
	            if (Object.prototype.hasOwnProperty.call(options, prop)) {
	                extended[prop] = options[prop];
	            }
	        }
	        return extended;
	    },

	    hasClass: function (elem, className) {
	        return new RegExp(' ' + className + ' ').test(' ' + elem.className + ' ');
	    },

	    addClass: function (elem, className) {
	        if (!this.hasClass(elem, className)) {
	            elem.className += ' ' + className;
	        }
	    },

	    removeClass : function (elem, className) {
	        var newClass = ' ' + elem.className.replace( /[\t\r\n]/g, ' ') + ' ';

	        if (this.hasClass(elem, className)) {
	            while (newClass.indexOf(' ' + className + ' ') >= 0 ) {
	                newClass = newClass.replace(' ' + className + ' ', ' ');
	            }
	            elem.className = newClass.replace(/^\s+|\s+$/g, '');
	        }
	    },

	    toggleClass:   function(elem, className){
	        if(this.hasClass(elem, className)){
	            console.log('has')
	            removeClass(elem, className);
	            return;
	        }
	        addClass(elem, className)
	    }
	}
}



},{}],"/Users/josep/Documents/work/puntoneutro/probes/utils/test.js":[function(require,module,exports){
var Utils = require('./');

var utils = Utils.call(this)

// var app = new APP();

// app.init();

var o1 = {
	test2:'josep'
}


var o2 = {
	test:'joan'
}

divtest = document.getElementById('test');

console.log(utils.extend(o1, o2));

utils.addClass(divtest, 'inactive');

// console.log("app is", APP);





},{"./":"/Users/josep/Documents/work/puntoneutro/probes/utils/index.js"}]},{},["/Users/josep/Documents/work/puntoneutro/probes/utils/test.js"]);
